### PR TITLE
feat: test commitment mining logic when forked

### PR DESCRIPTION
**Pull request by zzorum**

### CODE REVIEW

File: test/integration.test.ts [hunk 1]
```diff
@@ -523,8 +523,89 @@ describe("Work Hard Finance Integrated Test", function () {
...
+        alice.address
+      );
+    });
+    it("mined additional liquidity", async () => {
+      await goTo(day * 3);
+      const periphery = await workhard.getPeriphery(1);
+      const afterAliceVisionVal = await periphery.liquidityMining.mined(
+        alice.address
+      );
+      expect(afterAliceVisionVal.sub(beforeAliceVisionVal).isNegative()).is.eq(
```
Comment (0xMayflower) [inline test/integration.test.ts:540]: I recommend you to use `expect(a).to.be.gt(b)`

File: test/integration.test.ts [hunk 2]
```diff
@@ -523,8 +523,89 @@ describe("Work Hard Finance Integrated Test", function () {
...
+  describe("Alice does liquidity mining for 3 days for Carl's project", async () => {
+    let beforeAliceVisionVal: BigNumber;
+    runOnly(this, async () => {
+      const periphery = await workhard.getPeriphery(1);
+      beforeAliceVisionVal = await periphery.liquidityMining.mined(
+        alice.address
+      );
+    });
+    it("mined additional liquidity", async () => {
+      await goTo(day * 3);
```
Comment (0xMayflower) [inline test/integration.test.ts:535]: This line should be in the runOnly

File: test/integration.test.ts [hunk 3]
```diff
@@ -523,8 +523,89 @@ describe("Work Hard Finance Integrated Test", function () {
...
+        await forkedDAO.commit
+          .connect(bob)
+          .approve(forkedDAO.stableReserve.address, constants.MaxUint256);
+        await forkedDAO.baseCurrency
+          .connect(bob)
+          .approve(periphery.commitMining.address, constants.MaxUint256);
+        await forkedDAO.baseCurrency
+          .connect(bob)
+          .approve(forkedDAO.stableReserve.address, constants.MaxUint256);
+        console.log(
```
Comment (0xMayflower) [inline test/integration.test.ts:581]: Let's remove this logging ;)

**Review by 0xMayflower — CHANGES_REQUESTED** [review]

**Comment by zzorum** [comment]
@0xMayflower applied + additional testcase added. still in progress

File: test/integration.test.ts [hunk 4]
```diff
@@ -523,14 +530,246 @@ describe("Work Hard Finance Integrated Test", function () {
...
+          uri: "ipfs://MY_PROJECT_URL",
+        };
+        const projectsCountBefore = await project.projectsOf(1);
+        await project.connect(bob).createProject(1, projectMetadata.uri);
+        const projectsCountAfter = await project.projectsOf(1);
+        expect(projectsCountAfter.sub(projectsCountBefore)).to.be.eq(1);
+      });
+    });
+    describe("Raise fund", () => {
+      it("successfully add new funds", async () => {
```
Comment (0xMayflower) [inline test/integration.test.ts:649]: Recommend you to use this pattern :)
```
describe("add new funds", ()=> {
  runOnly(async() => {
    // transactions
  })
  it("statements", async () => {
    // expect statements.
  })
})
```

**Comment by 0xMayflower** [comment]
Merged them for the deployment preparation :) Please keep writing up the remaining cases!

**Pull request merged**